test/cql-pytest/conftest.py

```diff
@@ -93,9 +93,17 @@ def this_dc(cql):
 # and automatically deleted at the end. We use scope="session" so that all
 # tests will reuse the same keyspace.
 @pytest.fixture(scope="session")
-def test_keyspace(cql, this_dc):
+def test_keyspace(request, cql, this_dc):
     name = unique_name()
-    cql.execute("CREATE KEYSPACE " + name + " WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', '" + this_dc + "' : 1 }")
+    stmt = "CREATE KEYSPACE " + name + " WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', '" + this_dc + "' : 1 }"
+    if hasattr(request, "param"):
```

Comment:
The CQL which you generate here will fail on Cassandra and older Scylla. Please use the has_tablets fixture (which you already have) and if it's false, return the same keyspace.

It would be even better if you could make the **list** of parametrizations calculated: On Cassandra or older Scylla without tablets, you only run the test once. Just on Scylla with tablets you run it twice. I can't say I know how to do this, so I don't insist. It's not terrible that a few tests will needlessly run twice on Cassandra.